dll/win32/shell32/shlexec.cpp

```diff
@@ -2124,6 +2124,41 @@ static BOOL SHELL_execute(LPSHELLEXECUTEINFOW sei, SHELL_ExecuteW32 execfunc)
             *end = L'\0';
             lpFile = wfileName;
         }
+        else if (!PathIsDirectoryW(wszApplicationName) &&
+            !PathIsExeW(wszApplicationName) && 
+            (sei_tmp.fMask & SEE_MASK_INVOKEIDLIST) != SEE_MASK_INVOKEIDLIST &&
+            (sei_tmp.fMask & SEE_MASK_HASLINKNAME) != SEE_MASK_HASLINKNAME &&
+            (sei_tmp.fMask & SEE_MASK_FLAG_NO_UI) != SEE_MASK_FLAG_NO_UI)
+        {
+            /* If the executable name is not quoted, we have to use this search loop here,
+               that in CreateProcess() is not sufficient because it does not handle shell links. */
+            WCHAR buffer[MAX_PATH], xlpFile[MAX_PATH];
+            LPWSTR space, s;
+
+            LPWSTR beg = wszApplicationName/*sei_tmp.lpFile*/;
+            for(s = beg; (space = const_cast<LPWSTR>(strchrW(s, L' '))); s = space + 1)
+            {
+                int idx = space - sei_tmp.lpFile;
+                memcpy(buffer, sei_tmp.lpFile, idx * sizeof(WCHAR));
+                buffer[idx] = '\0';
+
+                /*FIXME This finds directory paths if the targeted file name contains spaces. */
```

Comment:
This does not explain why it's a fixme...

Reply:
Added comments showing what this section of code does. Removed this old comment because it no longer applies.